finish/src/main/java/io/openliberty/guides/inventory/model/InventoryList.java

```diff
@@ -12,32 +12,35 @@
 // end::copyright[]
 package io.openliberty.guides.inventory.model;
 
-import java.util.ArrayList;
 import java.util.List;
 import java.util.Properties;
+import java.util.concurrent.CopyOnWriteArrayList;
 
 public class InventoryList {
 
-    private List<SystemEntry> systems = new ArrayList<SystemEntry>();
-
+    private List<SystemEntry> systems = new CopyOnWriteArrayList<SystemEntry>();
```

Comment:
Do we need this?

Reply:
This is needed to fix the thread safe issue. There was another approach, which used the ArrayList, but then adding a lock for the get method. This will make the get method less efficient. Our application calls the get methods more often than others, so this approach was chosen. I presented both to Andrew, and he thinks this one is good.